test/plug/adapters/test/conn_test.exs

```diff
@@ -59,4 +59,21 @@ defmodule Plug.Adapters.Test.ConnTest do
     assert {:ok, params, _} = adapter.parse_req_multipart(state, 1_000_000, fn _ -> :ok end)
     assert params == %{"a" => "b", "c" => [%{"d" => "e"}, "f"]}
   end
+
+  test "use existing conn.host if exists" do
+    conn_with_host = conn(:get, "http://www.elixir-lang.org/")
+    assert "www.elixir-lang.org" == conn_with_host.host
```

Comment:
You won't get an "assignment" error (as you may expect) in Elixir as `=` is a matching construct. If you write `assert "foo" = bar`, then it will try to match the value of `bar` against `"foo"`; in this particular case, the match will only succeed if `bar` is `"foo"`, so doing `assert "foo" = bar` has very similar semantics to doing `assert bar == "foo"`. Long story short, you still probably want `assert conn_with_host == "www.elixir-lang.org"` here as it's the convention in most of Elixir's code base anyways :smile: